sympy/geometry/ellipse.py

```diff
@@ -140,7 +140,9 @@ def __new__(
                 "eccentricity" must not be None.'''))
 
         if eccentricity is not None:
-            if hradius is None:
+            if S(eccentricity).is_negative is True:
```

Comment:
There is already `eccentricity = sympify(eccentricity)` above and I'd want to know the reason why sympify is doubly needed here

Reply:
@sylee957 I was a bit confused about `sympify` and `S`, I got this now. Sorry for not seeing this. I have updated it.